### PR TITLE
feat: batch process for adding new GTFS feeds (Schedule + Realtime)

### LINKED ISSUE #16: Batch process for adding new GTFS feeds (Schedule + Realtime)
## **User Story**

As an API consumer, I need to access new GTFS and GTFS Realtime feeds that added to the Mobility Database Github repo so I can discover new data. 

## **Acceptance Criteria**

- A batch process that automatically stores new GTFS feeds from the Mobility Database Github repo that with all authentication types. Includes both GTFS Schedule and Realtime feeds.
- Batch process should include all the steps for unauthenticated GTFS feeds and GTFS realtime outlined in [this flowchart](https://miro.com/app/board/uXjVM86LB1s=/?moveToWidget=3458764558857335430&cot=14)
Public  Mobility Database API_ User Story Mapping - Flowchart for adding new feeds (1)

- Batch process should include all the steps for updating feeds outlined in [this flowchart](https://miro.com/app/board/uXjVM86LB1s=/?moveToWidget=3458764558860655921&cot=14)
Public  Mobility Database API_ User Story Mapping - Flowchart for updating feeds

## **Materials**

- [Current cronjob for storing latest datasets in the Mobility Database catalogs](https://github.com/MobilityData/mobility-database-catalogs/blob/main/.github/workflows/store_latest_datasets_cronjob.yml)

## **Considerations and Materials**

- [PR that added GTFS feed workflows to the Mobility Database catalogs repo](https://github.com/MobilityData/mobility-database-catalogs/pull/61)

### PR DESCRIPTION
**Pull request by cka-y**
**Summary:**
This update ensures GTFS feeds are seamlessly integrated into the Mobility PostgreSQL Database following their addition to the catalog repository. 
Closes #16

**Expected Behavior:** 
The process of adding a GTFS feed involves the following steps:
- For authenticated feeds, generate a key and store it in the `TECH: Keys & Token` vault. Place it under the `API Key — API_SOURCE_SECRETS` category in the `credential` item, using `mdb-<id>` as the key.
- Create a PR in the catalog repository to include the new feed, and merge it.
- The `db-update` action will trigger automatically across all environments, performing these tasks:
  - It reads the catalog CSV and executes the `populate.sh` script. This script is responsible for updating existing feeds if there are any changes and adding new feeds. It ensures no data alteration for unchanged feeds.
  - It updates the secrets in GCP that store authentication keys for feeds requiring authentication.

**Testing Tips:**
To effectively test this feature, focus on ensuring the `populate.sh` script does not alter data for feeds that haven't been updated. Follow these steps for a comprehensive test:
- Set up the environment variables file in `./config/.env.local` if it's not already present.
- Start a local instance of the database using `docker-compose --env-file ./config/.env.local up -d liquibase-test`.
- Run the `populate.sh` script within a Python venv (version 3.10): `./scripts/populate.sh "path to sources.csv" > populate.log`.
- Review the `populate.log` file. If you began with an empty database, it should display insert statements.
- Execute the script again using `./scripts/populate.sh "path to sources.csv" > populate_2.log`.
- Examine the `populate_2.log` file. If the `sources.csv` file remains unaltered, you should not see any new insert statements.
- To test further, add a new feed ID to the end of the `sources.csv` file and rerun the script. The log should now show multiple insert statements for the entities linked to the new feed, while existing feeds should remain unaffected.

Provide tips, procedures and sample files on how to test the feature.
Testers are invited to follow the tips AND to try anything they deem relevant outside the bounds of the testing tips. 

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [x] Add or update any needed documentation to the repo
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Review by davidgamez — APPROVED** [review]
💪 Awesome 💪

File: .github/workflows/db-update.yml [hunk 1]
```diff
@@ -24,99 +46,120 @@ on:
...
     - name: Upload log file for verification
-      if: inputs.RUN_POPULATE_SCRIPT == 'true'
       uses: actions/upload-artifact@v3
       with:
         name: populate.log
         path: populate.log
 
+  update-gcp-secret:
+    name: Update GCP Secrets
+#    if: ${{ github.event_name == 'repository_dispatch' || github.event_name == 'workflow_dispatch' }}
```
Comment (davidgamez) [inline .github/workflows/db-update.yml:127]: Do we need this conditional uncommented?

**Review by davidgamez — APPROVED** [review]
💪

File: .github/workflows/db-update.yml (hunk 1 above)
Comment (cka-y) [reply .github/workflows/db-update.yml]: Yes!

**Pull request merged**